src/neumann.jl

```diff
@@ -0,0 +1,58 @@
+# If face_value is ScalarValued, f should return scalar 
+# If face_value is VectorValued, f should return a vector
+struct Neumann{FV}      # cells and faces are a deconstructed
+    cells::Vector{Int}  # FaceIndex, to allow loop over cells 
+    faces::Vector{Int}  # with CellIterator 
+    face_values::FV     
+    field_name::Symbol
+    fun # f(x::Vec, t::Number, normal::Vec)->Union{<:Number, <:Vec}
+end
+
+function Neumann(field_name::Symbol, ∂Ω::Set{FaceIndex}, fv::FaceValues, f)
+    cells, faces = tuple((collect([face[j] for face in ∂Ω]) for j in 1:2)...)
+    return Neumann(cells, faces, fv, field_name, f)
+end
+
+function apply!(f::Vector{T}, nbc::Neumann, dh::DofHandler, time) where T
+    dofs = collect(dof_range(dh, nbc.field_name))
+    fe = zeros(T, length(dofs))
+    for (i, cell) in enumerate(CellIterator(dh, nbc.cells))
+        calculate_element_force!(fe, cell, nbc.face_values, nbc.faces[i], time, nbc.fun)
+        assemble!(f, view(celldofs(cell), dofs), fe)
+    end
+end
+
+function calculate_element_force!(fe::Vector, cell::CellIterator, fv::FaceValues, face_nr::Int, time, fun)
+    fill!(fe, 0)
+    reinit!(fv, cell, face_nr)
+    for q_point in 1:getnquadpoints(fv)
+        dΓ = getdetJdV(fv, q_point)
+        x = spatial_coordinate(fv, q_point, cell.coords)
+        n = getnormal(fv, q_point)
+        b = fun(x, time, n)
+        for i in 1:getnbasefunctions(fv)
+            δu = shape_value(fv, q_point, i)
+            fe[i] += δu ⋅ b * dΓ
+        end
+    end
+end
```

Comment:
What motivated me to include everything is that Dirichlet boundary conditions are currently very easy, but Neumann is not. 
Perhaps I'm mistaken here, but I thought this one should cover most standard cases (contrary to element routines, which are more specific to the physics). 
If you have time/possibility, could you share how this is not covering your case?

Reply:
I can for example have boundary conditions on a diffusion problem. Possibly that could be squeezed in to run with this, but in coupled problems you could also need several fields in the expression for the Neumann bcs, resulting in needing several `Values` objects (I believe this is the case e.g. for phase field fracture). And as Ferrite is more of a general toolbox than a mechanical FE program, we should also consider that people might be solving completely different differential equations - I'm not sure how exactly the boundary conditions that are arising from e.g. electro-magnetics or electro-chemically coupled problems look.

Reply:
Yes, by necessity this has to be tweaked to the particular weak form that you are solving (since a Neumann condition is just a part of the weak form). But depending on how you formulate it, (primal vs dual format) that will look different. So, to me, this starts to look like a "weak form library" but if that is the case, why only have the part of the weak form on the face and not the whole weak form in it?

Reply:
Thanks for the feedback guys!
I'll leave it open for now, working on making Neumann more convenient but independent of the weak form. 
Might turn into something useful, or not - let's see :)